udp/udp.go

```diff
@@ -68,19 +68,22 @@ func (s *Server) serve(listenAddr string) error {
 			return err
 		}
 
-		start := time.Now()
-
 		go func() {
-			response, action := s.handlePacket(buffer[:n], addr)
-			pool.GiveSlice(buffer)
+			start := time.Now()
+			response, action, err := s.handlePacket(buffer[:n], addr)
+			defer pool.GiveSlice(buffer)
```

Comment:
I didn't really want it to count in the timing, but wanted to keep it in the context of the `handlePacket` call.